template/deploy/helm/[[operator]]/templates/_helpers.tpl

```diff
@@ -74,3 +74,46 @@ Labels for Kubernetes objects created by helm test
 {{- define "operator.testLabels" -}}
 helm.sh/test: {{ include "operator.chart" . }}
 {{- end }}
+
+{{/*
+Include default resources or load resources from values if defined there.
+This is done with an if block instead of the default function because we need to pass anything
+retrieved from Values.resources to 'toYaml', but cannot do the same for the defaults we load.
+But if we do:
+  default (.Files.Get "resources/default_resources.yaml") (toYaml .Values.resources)
+then (toYaml .Values.resources) never is undefined, so the default value is effectively bypassed every time.
+*/}}
+{{- define "operator.resources" -}}
+{{- if .Values.resources }}
+{{- .Values.resources | toYaml }}
+{{- else }}
+{{- .Files.Get "resources/default_resources.yaml" }}
+{{- end }}
+{{- end }}
+
+{{/*
+Similar to operator.resources, but meant to be used for setting resources on daemonsets.
+As some operators have both, we wanted to be able to define different values.
+See comment on `operator.resources` for more details.
+*/}}
+{{- define "operator.controller.resources" -}}
```

Comment:
IMO, we should pick either this or the "bare" `.resources` across the whole platform.

Reply:
in that case I'd vote for keeping it the bare resources to avoid "cosmetic" (but strictly speaking breaking) changes througout all operators.

Reply:
I'm fine with that (arguably the controllers are the "primary" components), but don't we end up doing this anyway since both variants are new?

Reply:
`.resources` exists already and is picked up by the helm chart if set.

https://github.com/stackabletech/nifi-operator/blob/8f1797e1137207547bdcbddc38c122e48b50c0e2/deploy/helm/nifi-operator/values.yaml#L34